### ISSUE TITLE
`Comparing before and after` never finishes

### ISSUE THREAD

**Issue by SimonMarquis** [theapache64/dex-diff#1]
👋 Hi, we have a quite large app ~80MiB, and the process never finishes.
It is stuck at this step: `➡️ Comparing before and after... (this may take some time)`

I was wondering if there was a way to configure the max heapSize or CPU for this tool (RAM is probably the limiting factor)?

```
⚔️ dex-diff v0.0.6
🚀 Initialising...
➡️ Deleting old results (dex-diff-result)...
✅ Deleted old results
➡️ Decompiling before APK... (this may take some time)
✅ Decompiling before APK finished
➡️ Decompiling after APK... (this may take some time)
✅ Decompiling after APK finished
✅ Decompile finished (403417ms)
➡️ Comparing before and after... (this may take some time)
```

**Comment by theapache64** [comment]
Hi Simon,

80MiB sounds quite large.  how much time you waited for the comparison to
happen?
I just published 0.0.6's jar here ->
https://github.com/theapache64/dex-diff/releases/tag/0.0.6
You can try modifying the memory params like this
```
java -jar -Xms4096M -Xmx8192M dex-diff.main.jar  before.apk after.apk
com.my.app
```

Let me know if it worked or not.

**Comment by SimonMarquis** [comment]
Like a few hours 🙈
I'll try manually setting the heap size with the jar executable next week 👍

**Comment by SimonMarquis** [comment]
Allocating 16G seems to have make it work!
Though, heap size usage and CPU dropped during the last step, there might be something else throttling the process...

<img width="1222" alt="image" src="https://github.com/theapache64/dex-diff/assets/1921278/7ae5848a-1a58-4d6d-858d-29bee1290722">

**Comment by akexorcist** [comment]
Thanks to myself for finding this issue before waiting until it was done on my laptop 😂

With `-Xms8g` and `-Xmx16g` configurations and my APKs size is about 51MB, it took 105963ms (almost 2 minutes) for comparing.

**Comment by theapache64** [comment]
I just published `0.1.2` and it has Xms/x configs added; Closing this issue as there's no pending AIs on this; Feel free to reopen if you think otherwise guys

**Comment by theapache64** [comment]
@SimonMarquis the drop is probably happening after decompilation and the next step after is a single thread file-by-file comparison; maybe i should be make it multi-threaded and it should happen even faster 🤔
